### PR TITLE
Wildcard field bug fix for term queries. Wildcard syntax should not be supported in plain term queries.

### LINKED ISSUE #62081: Wildcard field term queries have false positive matches if they contain * or ? characters
The `term` query on wildcard fields delegates to the `wildCardQuery` method to do searching but this method assumes `*` or `?` characters have special meaning. 
This means the term query `foo?` mistakenly matches `foot`, `fool` etc rather than just docs with  `foo?`

### PR DESCRIPTION
**Pull request by markharwood**
Replaced the simple logic that delegated to wildcardQuery() method with dedicated exact-match query logic for term queries.
Added test.

Closes #62081

### CODE REVIEW

File: x-pack/plugin/wildcard/src/test/java/org/elasticsearch/xpack/wildcard/mapper/WildcardFieldMapperTests.java [hunk 1]
```diff
@@ -179,6 +179,41 @@ public void testTooBigQueryField() throws IOException {
         reader.close();
         dir.close();
     }
+    
+    
+    public void testTermQueryIgnoresWildcardSyntax() throws IOException {
+        // Test for bug https://github.com/elastic/elasticsearch/issues/62081
+        // where term queries honoured wildcard quwery syntax
```
Comment (jimczi) [inline x-pack/plugin/wildcard/src/test/java/org/elasticsearch/xpack/wildcard/mapper/WildcardFieldMapperTests.java:186]: I don't think we need the link on the issue. The test name is descriptive enough imo.

File: x-pack/plugin/wildcard/src/main/java/org/elasticsearch/xpack/wildcard/mapper/WildcardFieldMapper.java [hunk 2]
```diff
@@ -844,7 +845,36 @@ public Query existsQuery(QueryShardContext context) {
 
         @Override
         public Query termQuery(Object value, QueryShardContext context) {
-            return wildcardQuery(BytesRefs.toString(value), MultiTermQuery.CONSTANT_SCORE_REWRITE, context);
```
Comment (jimczi) [inline x-pack/plugin/wildcard/src/main/java/org/elasticsearch/xpack/wildcard/mapper/WildcardFieldMapper.java:847]: Isn't it the same problem with `prefix` query ? Maybe we can just escape `?` and `*` in the value and continue to use the `wildcardQuery` ?

**Review by jimczi — COMMENTED** [review]
I left one question.

File: x-pack/plugin/wildcard/src/main/java/org/elasticsearch/xpack/wildcard/mapper/WildcardFieldMapper.java (hunk 2 above)
Comment (markharwood) [reply x-pack/plugin/wildcard/src/main/java/org/elasticsearch/xpack/wildcard/mapper/WildcardFieldMapper.java:847]: Good point. I'll try that approach.

**Review by jimczi — APPROVED** [review]
LGTM

**Pull request merged**